Ilya/NetflixRoulette/netflix-roulette/src/redux/moviesSagas.ts

```diff
@@ -0,0 +1,86 @@
+import {
+    MoviesActionTypes,
+    RequestMoviesAction,
+    requestMoviesSuccess,
+    requestMoviesError,
+    RequestMovieByIdAction,
+    requestMovieByIdSuccess,
+    requestMovieByIdError,
+
+    RequestMovieDetailsDataAction,
+    requestMovieById,
+    requestMovies,
+
+} from "./moviesActions";
+import { all, call, put, select, takeLatest } from 'redux-saga/effects';
+import queryString from "query-string";
+
+function* requestMoviesSaga(action: RequestMoviesAction) {
+    let search: string = action.payload.search;
+    let offset = 0;
+
+    if (search === "") {
+        search = "sortBy=release_date&offset=0";
+    } else {
+        let parsed = queryString.parse(search);
+        if ('page' in parsed) {
+            offset = (Number(parsed.page) - 1) * 9;
+            delete parsed.page;
+        }
+        parsed.offset = String(offset);
+        search = queryString.stringify({ ...parsed });
+    }
+
+    try {
+        const films = yield call(() => {
+            return (fetch(`https://reactjs-cdp.herokuapp.com/movies?sortOrder=desc&${search}&limit=9`)
+                .then(response => response.json()));
+        });
+
+        yield put(requestMoviesSuccess(films.data, films.total));
+
+    } catch (error) {
+        yield put(requestMoviesError());
+    }
+}
+
+function* requestMovieByIdSaga(action: RequestMovieByIdAction) {
+
+    try {
+        const film = yield call(() => {
+            return (fetch(`https://reactjs-cdp.herokuapp.com/movies/${action.payload.id}`)
+                .then(response => response.json()));
+        });
+        yield put(requestMovieByIdSuccess(film));
+    } catch (error) {
+        yield put(requestMovieByIdError());
+    }
+}
+
+//------------------ FetchMovieDetailsData
+function* requestMovieDetailsDataSaga(action: RequestMovieDetailsDataAction) {
+    yield* requestMovieByIdSaga(requestMovieById(action.payload.id));
+    const filmGenre = yield select((store) => store.moviesStore.movie.genres[0]);
```

Comment:
yield* requestMovieByIdSaga will return you success action, so you can remove yield select bellow and write like this. 
` const yourActionPayload =  yield* requestMovieByIdSaga(...);`